frappe/core/doctype/communication/communication.py

```diff
@@ -463,7 +464,124 @@ def update_first_response_time(parent, communication):
 				parent.db_set("first_responded_on", first_responded_on)
-			parent.db_set("first_response_time", round(time_diff_in_seconds(first_responded_on, parent.creation), 2))
+			first_response_time = calculate_first_response_time(parent, get_datetime(first_responded_on))
+			parent.db_set("first_response_time", first_response_time)
+
+def calculate_first_response_time(issue, first_responded_on):
+	issue_creation_date = issue.creation
+	issue_creation_time = get_time_in_seconds(issue_creation_date)
+	first_responded_on_in_seconds = get_time_in_seconds(first_responded_on)
+
+	if issue_creation_date.day == first_responded_on.day:
+
+		start_time, end_time = get_working_hours(issue_creation_date, issue)
+
+		# issue creation and response on the same day during working hours
+		if is_during_working_hours(issue_creation_date, issue) and is_during_working_hours(first_responded_on, issue):
+			return get_elapsed_time(first_responded_on, issue_creation_date)
+
+		# issue creation is during working hours, but first response was after working hours
+		elif is_during_working_hours(issue_creation_date, issue):
+			return get_elapsed_time(end_time, issue_creation_time)
+
+		# issue creation was before working hours but first response is during working hours
+		elif is_during_working_hours(first_responded_on, issue):
+			return get_elapsed_time(start_time, first_responded_on_in_seconds)
+
+		# both issue creation and first response were after working hours
+		else:
+			return 0.0
+
+	else:
+		# response on the next day
+		if first_responded_on.day - issue_creation_date.day == 1:
+			first_response_time = 0
+		else:
+			first_response_time = calculate_initial_frt(issue, issue_creation_date, first_responded_on.day - issue_creation_date.day - 1)
+
+		# time taken on day of issue creation
+		start_time, end_time = get_working_hours(issue_creation_date, issue)
+
+		if is_during_working_hours(issue_creation_date, issue):
+			first_response_time += get_elapsed_time(issue_creation_time, end_time)
+
+		elif is_before_working_hours(issue_creation_date, issue):
+			first_response_time += get_elapsed_time(start_time, end_time)	
+		else:
+			first_response_time += 0
+
+		# time taken on day of first response
+		start_time, end_time = get_working_hours(first_responded_on, issue)
+
+		if is_before_working_hours(first_responded_on, issue):
+			first_response_time += 0
+		elif is_during_working_hours(first_responded_on, issue):
+			first_response_time += get_elapsed_time(start_time, first_responded_on_in_seconds)
+		else:
+			first_response_time += get_elapsed_time(start_time, end_time)
+
+		return first_response_time
+
+def get_time_in_seconds(date):
+	return timedelta(hours=date.hour, minutes=date.minute, seconds=date.second)
+
+def get_working_hours(date, issue):
+	weekday = frappe.utils.get_weekday(date)
+	work_days = frappe.get_doc("Service Level Agreement", issue.service_level_agreement).support_and_resolution
```

Comment:
Does this mean Communication will stop working without ERPNext installed?

Reply:
That hadn't occurred to me, thanks for pointing it out. I've moved the code specific to Issue to issue.py.